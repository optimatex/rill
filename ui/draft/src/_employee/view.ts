export {
  React,
  useState,
  useCallback,
  css,
  styled,
  createGlobalStyle,
} from '@rill/view';
