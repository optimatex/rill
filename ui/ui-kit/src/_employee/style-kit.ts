export {
  // appLayout
  appLayoutCss,
  screenCss,
  sidebarColumnCss,
  flatColumnCss,
  mainColumnCss,
  // layout
  listCss,
  // Brand
  logCss,
  dualLogoCss,
  // Controller
  trivialAppHeaderCss,
  // _menu
  menuWrapCss,
  menuItemCss,
  // Image
  imageCss,
  imagesListCss,
  imagesGridCss,
  imagesLayoutItemCss,
  imageCropperCss,
  // Tags
  tag,
  tagsList,
  // Ui-primal
  // _grid
  gridRowContainerCss,
  gridrillRowContainerCss,
  gridColumnCss,
  // _card
  cardContainerCss,
  postContainerCss,
  postTitleCss,
  panelCss,
  // _buttons
  buttonCss,
  buttonActiveCss,
  buticonCss,
  // _form
  formContainerCss,
  formFieldFlatCss,
  formFieldColumnCss,
  formFieldErrorCss,
  formLabelTextCss,
  inputCss,
  toggleContainerCss,
  toggleContainerActiveCss,
  toggleRoundCss,
  toggleRoundActiveCss,
  // _article
  paragraphCss,
  nameCss,
  richContentWrapCss,
  // _table
  tableContainerCss,
  tableHeadCss,
  tableRowCss,
  tableColumnCss,
  // Unknown
  // _object
  stringValueCss,
  numberValueCss,
  booleanValueCss,
  emptyValueCss,
  arrayWrapCss,
  arrayItemCss,
  objectKeyCss,
  objectWrapCss,
  objectPairCss,
  // Unsorted
  lightIndicatorCss,
  popoverCss,
  pipeCss,
  hintableLogCss,
  fixedPanelCss,
} from '@rill/style-kit';
