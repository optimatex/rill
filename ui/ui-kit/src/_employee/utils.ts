export {
  toPairs,
  find,
  has,
  path,
  prop,
  curry,
  assoc,
  lensProp,
  view,
  set,
} from '@rill/utils';
