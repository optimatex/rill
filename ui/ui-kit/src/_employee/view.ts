export {
  React,
  useState,
  useCallback,
  // styled-components
  styled,
  // Lens
  // getTestId,
  // Types
  FC,
  RC,
  ReactNode,
  LegacyRef,
  // PropTypes
  IContainerProps,
  // Event types
  FormEvent,
  MouseEvent,
} from '@rill/view';
