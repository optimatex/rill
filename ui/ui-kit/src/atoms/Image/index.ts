export * from './Image';
export * from './ImagesLayout';
