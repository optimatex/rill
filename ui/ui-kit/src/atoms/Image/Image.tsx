import { styled } from '_employee/view';
import { imageCss } from '_employee/style-kit';

export const Image = styled.img`
  ${imageCss}
`;
