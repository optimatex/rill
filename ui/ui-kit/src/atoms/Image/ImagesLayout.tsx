import { React, styled } from '_employee/view';
import {
  imagesListCss,
  imagesGridCss,
  imagesLayoutItemCss,
} from '_employee/style-kit';

import { IImage, LayoutTypes } from '@types';
import { Image } from './Image';

// let largeCount: number, lastSmallIndex: number;

//   if (layout === 'grid') {
//     // The algorithm that defines outside images which need to be bigger
//     largeCount = images.length % 3;
//     lastSmallIndex = images.length - largeCount - 1;
//   }

// const isBigger = index > lastSmallIndex;
//       const size = defineSize(isBigger, layout, largeCount);

interface ILayoutProps {
  images: IImage[];
  layout: LayoutTypes;
}
const defaultProps = {
  layout: 'list',
};

const List = styled.div`
  ${imagesListCss}
`;
const Grid = styled.div`
  ${imagesGridCss}
`;
const Item = styled.div`
  ${imagesLayoutItemCss}
`;

export const ImagesLayout = ({ layout, images }: ILayoutProps) => {
  const Layout = layout === 'grid' ? Grid : List;

  return (
    <Layout data-testid="images-layout">
      {images.map((item: IImage) => (
        <Item key={item.src}>
          <Image src={item.src} />
        </Item>
      ))}
    </Layout>
  );
};
ImagesLayout.defaultProps = defaultProps;
