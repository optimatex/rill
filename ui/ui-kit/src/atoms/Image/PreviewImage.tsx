import { React } from '_employee/view';

// import {Link} from '../Link'
import { Image } from './Image';

interface IProps {
  src: string;
}
// <Link to={`${props.src}?image=`}>
export const PreviewImage = (props: IProps) => <Image src={props.src} />;
