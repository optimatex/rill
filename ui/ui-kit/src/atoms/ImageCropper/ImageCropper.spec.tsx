import { render, mockImageURL_1 } from 'test';
import { React } from '_employee/view';
import { Image } from '../Image';
import { ImageCropper } from '.';

describe('ImageCropper', () => {
  it('should render without errors', () => {
    const { container } = render(
      <ImageCropper>
        <Image src={mockImageURL_1} />
      </ImageCropper>,
    );
    const el = container.querySelector('div');

    expect(el).toMatchSnapshot();
  });
});
