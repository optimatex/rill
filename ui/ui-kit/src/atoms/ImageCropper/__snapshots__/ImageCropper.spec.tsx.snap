// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageCropper should render without errors 1`] = `
.c1 {
  object-fit: cover;
  max-width: 100%;
  display: block;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: none;
  -ms-flex: none;
  flex: none;
  height: 30px;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  margin: 0 auto;
  width: 30px;
  overflow: hidden;
  border-radius: 50px;
}

<div
  class="c0"
>
  <img
    class="c1"
    src="url-1.jpg"
  />
</div>
`;
